src/main/java/frc/robot/subsystems/SubsystemManager.java

```diff
@@ -8,22 +8,44 @@
 
 public class SubsystemManager {
     private DriveSubsystem driveSubsystem;
-
+    private NavXSubSystem navXSubSystem;
+    private VisionSubsystem visionSubsystem;
+    private ArmSubsystem armSubsystem;
+    
     public SubsystemManager() {
     }
 
     public DriveSubsystem getDriveSubsystem() {
         return driveSubsystem;
     }
 
+    public NavXSubSystem getNavXSubSystem() {
+        return navXSubSystem;
+    }
+    public VisionSubsystem getVisionSubsystem() {
+        return visionSubsystem;
+    }
 
+    public ArmSubsystem getArmSubsystem(){
+        return armSubsystem;
+    }
+    
     public void initAll() {
-        driveSubsystem = new DriveSubsystem();
-
+        navXSubSystem  = new NavXSubSystem();
+        driveSubsystem = new DriveSubsystem(getNavXSubSystem());
```

Comment:
NO! we dont want subsystems holding references to each other-- this produces hard to debug flows. see dc_pose_updates